pandas/tests/indexes/datetimes/methods/test_delete.py

```diff
@@ -132,9 +132,9 @@ def test_delete_slice2(self, tz, unit):
         assert result.freq == expected.freq
         assert result.tz == expected.tz
 
-        # reset freq to None
+        # reset freq
         result = ts.drop(ts.index[[1, 3, 5, 7, 9]]).index
-        expected = dti[::2]._with_freq(None)
+        expected = dti[::2]._with_freq("infer")
```

Comment:
the "infer" here should be unnecessary if dti already has a freq

Reply:
removed `_with_freq("infer")`